torchmetrics/functional/classification/calibration_error.py

```diff
@@ -0,0 +1,123 @@
+
+import torch
+from torch import Tensor, tensor
+from torch.nn import functional as F
+
+from torchmetrics.utilities.checks import _input_format_classification
+from torchmetrics.utilities.enums import DataType
+
+
+def _ce_compute(
+    confidences: Tensor, accuracies: Tensor, bin_boundaries: Tensor, norm: str = "l1", debias: bool = False
+) -> Tensor:
+
+    conf_bin = torch.zeros_like(bin_boundaries)
+    acc_bin = torch.zeros_like(bin_boundaries)
+    prop_bin = torch.zeros_like(bin_boundaries)
+    for i, (bin_lower, bin_upper) in enumerate(zip(bin_boundaries[:-1], bin_boundaries[1:])):
+        # Calculated confidence and accuracy in each bin
+        in_bin = confidences.gt(bin_lower.item()) * confidences.le(bin_upper.item())
+        prop_in_bin = in_bin.float().mean()
+        if prop_in_bin.item() > 0:
+            acc_bin[i] = accuracies[in_bin].float().mean()
+            conf_bin[i] = confidences[in_bin].mean()
+            prop_bin[i] = prop_in_bin
+
+    if norm == "l1":
+        ce = torch.sum(torch.abs(acc_bin - conf_bin) * prop_bin)
+        print(ce)
+    elif norm == "max":
+        ce = torch.max(torch.abs(acc_bin - conf_bin))
+    elif norm == "l2":
+        ce = torch.sum(torch.pow(acc_bin - conf_bin, 2) * prop_bin)
+        # NOTE: debiasing is disabled in the wrapper functions. This implementation differs from that in sklearn.
+        if debias:
+            # the order here (acc_bin - 1 ) vs (1 - acc_bin) is flipped from
+            # the equation in Verified Uncertainty Prediction (Kumar et al 2019)/
+            debias_bins = (acc_bin * (acc_bin - 1) * prop_bin) / (prop_bin * accuracies.size()[0] - 1)
+            ce += torch.sum(torch.nan_to_num(debias_bins))  # replace nans with zeros if nothing appeared in a bin
+        ce = torch.sqrt(ce) if ce > 0 else torch.tensor(0)
+    else:
+        raise ValueError(f"Norm {norm} is not supported. Please select from l1, l2, or max. ")
+    return ce
+
+
+def _ce_update(preds: Tensor, target: Tensor) -> Tuple[Tensor, Tensor]:
+    _, _, mode = _input_format_classification(preds, target)
+
+    if mode == DataType.BINARY:
+        confidences, accuracies = preds, target
+    elif mode == DataType.MULTICLASS:
+        confidences, predictions = preds.max(dim=1)
+        accuracies = predictions.eq(target)
+    elif mode == DataType.MULTIDIM_MULTICLASS:
+        # reshape tensors
+        # for preds, move the class dimension to the final axis and flatten the rest
+        confidences, predictions = torch.transpose(preds, 1, -1).flatten(0, -2).max(dim=1)
+        # for targets, just flatten the target
+        accuracies = predictions.eq(target.flatten())
+    else:
+        raise ValueError(
+            f"Calibration error is not well-defined for data with size {preds.size()} and targets {target.size()}"
+        )
+
+    return confidences, accuracies
+
+
+def calibration_error(preds: Tensor, target: Tensor, n_bins: int = 15, norm: str = "l1"):
+    """
```

Comment:
this needs to be a raw string for the math to compile correctly
```suggestion
    r"""
```